src/app/components/Inside/Player/VolumeControl.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import Mousetrap from 'mousetrap';
+import withMK from '../../../hoc/withMK';
+import styles from './Player.scss';
+import { volumeUp, volumeDown } from '../../../services/MusicPlayerApi';
+
+class VolumeControl extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      volume: props.mk.instance.player.volume,
+    };
+
+    this.getVolumeIconClasses = this.getVolumeIconClasses.bind(this);
+    this.handleVolumeChange = this.handleVolumeChange.bind(this);
+    this.toggleVolume = this.toggleVolume.bind(this);
+  }
+
+  componentDidMount() {
+    // Volume controls (VOLUME UP)
+    Mousetrap.bind(
+      'up',
+      e => {
+        e.preventDefault();
+        this.setState({
+          volume: volumeUp(),
+        });
+      },
+      'keydown'
+    );
+
+    // Volume controls (VOLUME DOWN)
+    Mousetrap.bind(
+      'down',
+      e => {
+        e.preventDefault();
+        this.setState({
+          volume: volumeDown(),
+        });
+      },
+      'keydown'
+    );
+  }
+
+  getVolumeIconClasses() {
+    const { volume } = this.props.mk.instance.player;
+
+    if (volume === 0) {
+      return 'fas fa-times';
+    }
+
+    if (volume < 0.3) {
+      return 'fas fa-volume-off';
+    }
+
+    if (volume < 0.6) {
+      return 'fas fa-volume-down';
+    }
+
+    return 'fas fa-volume-up';
+  }
+
+  handleVolumeChange(e) {
```

Comment:
Lets add an `onVolumeChange` function that takes as a parameter the volume, use it in `handleVolumeChange` and in the various `Mousetrap.bind`

Reply:
Wouldn't `onVolumeChange` be reserved for passing a function as a callback as a param to the component? Sounds like an event that this would call, not used for internal volume change logic itself

Reply:
What name do you suggest ?

Reply:
even just changeVolume() would be better

Reply:
and perhaps rename handleVolumeChange() to handleVolumeBar() or something similar?

Reply:
Not a big fan

![https://i.ebayimg.com/images/g/btMAAOSwj0NUcpqL/s-l1600.jpg](fan)

Reply:
Okay for the edit

Reply:
Open to suggestions, but a prefix of `on` implies it's a callback function for an event